### PR TITLE
added fake service implementation

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: certificatetprtest/service.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package certificatetprtest
```
Comment (xh3b4sd) [inline certificatetprtest/service.go:1]: Here is the new fake implementation of the certificatetpr service being implemented. The naming and concept aligns with `microloggertest`.

File: certificatetprtest/service_test.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	"github.com/giantswarm/certificatetpr"
+)
+
+func Test_Fake_Service_SearcherInterface(t *testing.T) {
+	newService := NewService()
+	_, ok := interface{}(newService).(certificatetpr.Searcher)
+	if !ok {
+		t.Fatal("fake service does not implement Searcher interface")
+	}
+}
```
Comment (xh3b4sd) [inline certificatetprtest/service_test.go:15]: This test makes sure the fake implementation implements the `Searcher` interface, which is the only interface being defined in the certificatetpr package.

File: service.go [hunk 3]
```diff
@@ -20,25 +20,26 @@ const (
 	WatchTimeOut = 90 * time.Second
 )
 
-// Config represents the configuration used to create a service.
-type Config struct {
+// ServiceConfig represents the configuration used to create a certificate TPR
+// service.
+type ServiceConfig struct {
```
Comment (xh3b4sd) [inline service.go:25]: Here happened some renaming which makes the naming more correct based on our current standards.

File: spec.go [hunk 4]
```diff
@@ -1,5 +1,9 @@
 package certificatetpr
 
+type Searcher interface {
+	SearchCerts(clusterID string) (AssetsBundle, error)
+}
```
Comment (xh3b4sd) [inline spec.go:5]: Here is the new interface packages can use to be able to mock out the certificate watcher. This happens for instance in the kvm-operator. See https://github.com/giantswarm/kvm-operator/pull/150.

File: service.go [hunk 5]
```diff
@@ -20,25 +20,26 @@ const (
...
 		// Dependencies.
 		K8sClient: nil,
 		Logger:    nil,
 	}
 }
 
-// New creates a new configured service.
-func New(config Config) (*Service, error) {
+// NewService creates a new configured certificate TPR service.
+func NewService(config ServiceConfig) (*Service, error) {
```
Comment (kopiczko) [inline service.go:42]: Why `NewService` is better than `New`?

File: certificatetprtest/service.go (hunk 1 above)
Comment (kopiczko) [reply certificatetprtest/service.go:1]: Might make sense to make this a recorder, depending on your further needs. Like in `httptest.ResponseRecorder` https://golang.org/pkg/net/http/httptest/#ResponseRecorder

**Review by kopiczko — APPROVED** [review]

File: service.go (hunk 5 above)
Comment (xh3b4sd) [reply service.go:42]: This is the `certificatetpr` repository. Using `New` means creating a new certificate TPR. What we are creating here is a certificate TPR service that is e.g. able to search for certificates. So `NewService` is more appropriate here.

File: certificatetprtest/service.go (hunk 1 above)
Comment (xh3b4sd) [reply certificatetprtest/service.go:1]: Interesting idea but I do not see a need for it now. Lets iterate here and see how it goes.

**Pull request merged**